x-pack/plugins/enterprise_search/public/applications/app_search/components/result/result.tsx

```diff
@@ -33,6 +43,19 @@ export const Result: React.FC<Props> = ({ result, showScore }) => {
     [result]
   );
   const numResults = resultFields.length;
+  const typeForField = (fieldName: string) => {
+    if (schemaForTypeHighlights) return schemaForTypeHighlights[fieldName];
+  };
+
+  const conditionallyLinkedArticle = (children: React.ReactNode) => {
+    if (!shouldLinkToDetailPage)
+      return <article className="appSearchResult__content">{children}</article>;
+    return (
+      <ReactRouterHelper to={getDocumentDetailRoute(resultMeta.engine, resultMeta.id)}>
+        <a className="appSearchResult__content">{children}</a>
```

Comment:
So I was testing this in a screenreader, and was today years old when I found that apparently content in `<a>` or `<button>` tags causes screen readers to read out EVERYTHING inside that content instead of making them separately navigable like they were before 🤦‍♀️

<img width="625" alt="" src="https://user-images.githubusercontent.com/549407/102551623-a8f5d800-4074-11eb-958a-bd32af891e70.png">

The above sadly sounds like almost total gibberish. I checked with the folks in the accessibility Slack channel and apparently this is just default behavior for better or for worse, but it means I need to rethink my recommendation of wrapping everything in an `<a>` tag.

I have some ideas for a workaround that I ran by Davey that involves rethinking how we arrange our buttons. It's somewhat complex and follows what EUI does with their EuiCard component - I won't ask you to make those changes as part of those PR, but I'd like to open a separate follow-up PR or even PR your PR if you're cool with that. Definitely want your thoughts on it first!

Reply:
I see your point. I think we should look for a quick and easy fix though. I'm feeling like we've already spent too much time on this component.

Reply:
This could also be helpful because since the entire thing is now clickable, it makes it so you can't highlight and copy text as-is.

Reply:
You also need to leave room for a an additional button, which will be used for curations, when we get there.

Reply:
Like you could potentially opt for a potentially simpler solution that does something like adds the id as a link, or something like that, rather than re-arranging the entire component.

Reply:
Is it possible to just add an `aria-label` to the anchor as-is?

Reply:
> Is it possible to just add an aria-label to the anchor as-is?

That was my instinct at first as well for a quick solution - I checked with Michail and is apparently an "OK" but not great solution. We're essentially striking a balance in trade-offs between keyboard/screen reader users (readability, predictable link behavior) and mouse users (large & predictable link hitbox), so there's no one perfect solution here, only a 'best compromise'.

The issue with the aria-label suggestion I was originally going to make is that it essentially hides information from screen reader users, which isn't great, and it seems that it's preferred that we show too much information (even if it's hard to parse) over less information.

You can check out the full [Slack convo](https://elastic.slack.com/archives/C7CSV2REU/p1608227169127700) here if you like - I do have a working solution (https://github.com/jasonstoltz/kibana/compare/documents-view-5...constancecchen:result-a11y-enhancements) based on Davey's screenshot that only took a few hours to spin up, so hopefully it shouldn't require any more dev time than me updating tests for it. I can post screenshots and notes in a follow-up PR to it. LMK if you'd like to discuss this synchronously to clear anything up!